### ISSUE TITLE
Return complex field on Select statement

### ISSUE THREAD

**Issue by rafamerlin** [ElasticLINQ/ElasticLINQ#110]
I have a bigger document and I would like to return just 2 fields for example the suburb and the name, however, suburb is inside another object called address.

I can make it work by sending this to ES (I'm using version 1.4.5)

```
{
	"fields": ["address.suburb", "name"]
}
```

However, I can't make it work on ElasticLinq.
I believe this is doable and I'm just querying it incorrectly. 

If I have this classes (for example, my ES document as mentioned before is much bigger) :
```
public class Customer {
    public string Name {get;set;}
    public bool Active {get;set;}
    public Address Address {get;set;}
}

public class Address {
    public string Street {get;set;}
    public string Suburb {get;set;}
}
```

How could I select it on ElasticLinq?

I've tried this and many other similar variations, but it does not generate the proper query and returns a bad request.

```
var result = context.Query<Customer>
                   .Select(x => new {Name = x.Name, Suburb = x.Address.Suburb});
```
I've even tried to change Address to IQueryable and select inside of it (similar to the Any examples some people use to query nested itens in the Where clause) but I couldn't make it work either.

Is this doable at all in the current version?

Cheers

**Comment by damieng** [comment]
Can you post the bad query it generates here?

**Comment by rafamerlin** [comment]
Hey Damien, thanks for replying.
For:
```
var result = context.Query<Customer>
                   .Select(x => new {Name = x.Name, Suburb = x.Address.Suburb})
                   .Take(10).ToList();
```
The generated code is:
```
{"fields":["name","address"],"size":10,"timeout":"10s"}
```
And I get the leaf error in this request:
`nested: ElasticsearchIllegalArgumentException[field [address] isn't a leaf field]` 

I managed to work-around the issue by doing this:

```
public class CustomerResult {
    public string Name {get;set;}
    public bool Active {get;set;}
    [JsonProperty(PropertyName = "address.suburb")]
    public string Suburb {get;set;}
}
```

And my query:

```
var result = context.Query<CustomerResult>
                   .Select(x => new {Name = x.Name, Suburb = x.Suburb})
                   .Take(10).ToList();
```

My only concern is how could I deal with the same situation if instead of one address per customer if I had an array of addresses. Maybe there's a better way of doing it.

Thanks a lot for all the hard work you've put in this project.

**Comment by damieng** [comment]
Okay, error found and addressed for the single-property scenario.  Not sure what syntax we'd use for nested array properties besides the usual aggregate functions/last/first nor what query it would generate. Should probably create a separate issue for that.

**Comment by damieng** [comment]
Shipped in 1.4.3 now on NuGet.

**Comment by rafamerlin** [comment]
Thanks a lot Damien, I'll test it.
